nep-23.mediawiki

```diff
@@ -0,0 +1,331 @@
+|-
+| closewallet         || 
+|-
+| dumpprivkey         || -302
+|-
+| findstates          || -102, -606
+|-
+| getapplicationlog   || -105
+|-
+| getblock            || -101, -109
+|-
+| getblockhash        || -109
+|-
+| getblockheader      || -101, -109
+|-
+| getcandidates       ||
+|-
+| getcommittee        ||
+|-
+| getconnectioncount  ||
+|-
+| getcontractstate    || -102
+|-
+| getnativecontracts  ||
+|-
+| getnep11balances    ||
+|-
+| getnep11properties  || -608
+|-
+| getnep11transfers   ||
+|-
+| getnep17balances    ||
+|-
+| getnep17transfers   ||		
+|-
+| getnewaddress       || -302
+|-
+| getnextblockvalidators||
+|-
+| getpeers 	      ||
+|-
+| getproof            || -102, -104, -606
+|-
+| getrawmempool       || 			
+|-
+| getrawtransaction   || -103
+|-
+| getstate            || -102, -606
+|-
+| getstateheight      ||
+|-
+| getstateroot        || -106
+|-
+| getstorage          || -102, -104
+|-
+| gettransactionheight|| -103
+|-
+| getunclaimedgas     ||
+|-
+| getversion          ||
+|-
+| getwalletbalance    || -302
+|-
+| getwalletunclaimedgas|| -302
+|-
+| invokecontractverify|| -102, -512
+|-
+| invokefunction      ||
+|-
+| invokescript        ||
+|-
+| importprivkey       || -302
+|-
+| listaddress         || -302
+|-
+| listplugins 	      ||
+|-
+| openwallet          || -302, -303, -304
+|-
+| sendfrom            || -300, -301, -302
+|-
+| sendmany            || -300, -301, -302
+|-
+| sendtoaddress       || -300, -301, -302
+|-
+| sendrawtransaction  || -500, -501, -502, -503, -504, -505, -506, -507, -508, -509, -510, -511
+|-
+| submitblock         || -500, -501, -502, -503, -504, -505, -506, -507, -508, -509, -510, -511
+|-
+| submitoracleresponse|| -508, -602, -603, -604, -605
+|-
+| terminatesession    || -107, -601
+|-
+| traverseiterator    || -107, -108, -601
+|-
+| validateaddress     ||
+|-
+| verifyproof         || -606, -607
+|}
+
+
+==Rationale==
+
+A large set of codes and behavior nuances is directly inhereted from current
+implementations. New ones are added where clients need more details to
+properly interpret server response. The most important changes are in the
+verification group, specific errors allow clients to adjust the behavior
+accordingly: fail, retry or wait.
+
+==Backwards Compatibility==
+
+* -100 code is gone completely, it's returned for a very wide variety of different problems at the moment, so the code itself shouldn't matter much
+* -101 and -102 errors change their meaning, they're only used in <code>invokecontractverify</code> call which is not very popular
+* -300 and -301 wallet-related errors are completely compatible
+* -400 error is now -302 or -600 (depending on its meaning) which may affect some applications
+* -500 error has the same meaning and is compatible
+* new codes can affect client applications that expect only old ones
+
+==References==
+
+* [https://www.jsonrpc.org/specification JSON-RPC 2.0 Specification]
+
+==Implementation==
+
+<TBD>
```

Comment:
Maybe add some link to the implementation? We have two implementations in fact, one in C# and one in Go.

Reply:
Do you have the links?

Reply:
Resolved.